components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpointTest.java

```diff
@@ -556,8 +548,280 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
+            checkErrorCode =false;
+        }
+
+        mockStatic(OAuth2Util.OAuthURL.class);
+        when(OAuth2Util.OAuthURL.getOAuth2ErrorPageUrl()).thenReturn(ERROR_PAGE_URL);
+
+        OAuth2ClientValidationResponseDTO validationResponseDTO = new OAuth2ClientValidationResponseDTO();
+        validationResponseDTO.setValidClient(clientValid);
+        validationResponseDTO.setCallbackURL(APP_REDIRECT_URL);
+        if (!clientValid) {
+            validationResponseDTO.setErrorCode(OAuth2ErrorCodes.INVALID_REQUEST);
+            validationResponseDTO.setErrorMsg("client is invalid");
+        }
+        validationResponseDTO.setPkceMandatory(supportPlainPkce);
+        validationResponseDTO.setPkceSupportPlain(supportPlainPkce);
+        when(oAuth2Service.validateClientInfo(anyString(), anyString())).thenReturn(validationResponseDTO);
+
+        final String[] redirectUrl = new String[1];
+
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                String key = (String) invocation.getArguments()[0];
+                redirectUrl[0] = key;
+                return null;
+            }
+        }).when(httpServletResponse).sendRedirect(anyString());
+
+        Response response = oAuth2AuthzEndpoint.authorize(httpServletRequest, httpServletResponse);
+
+        if (response != null) {
+            assertEquals(response.getStatus(), HttpServletResponse.SC_FOUND, "Unexpected HTTP response status");
+
+            MultivaluedMap<String, Object> responseMetadata = response.getMetadata();
+            assertNotNull(responseMetadata, "Response metadata is null");
+
+            assertTrue(CollectionUtils.isNotEmpty(responseMetadata.get(HTTPConstants.HEADER_LOCATION)),
+                    "Location header not found in the response");
+            String location = (String) responseMetadata.get(HTTPConstants.HEADER_LOCATION).get(0);
+            assertTrue(location.contains(expectedLocation), "Unexpected redirect url in the response");
+
+            if (checkErrorCode) {
+                assertTrue(location.contains(OAuth2ErrorCodes.INVALID_REQUEST), "Expected error code not found in URL");
+            }
+        } else {
+            assertNotNull(redirectUrl[0], "Response not redirected to outside");
+        }
+    }
+
+    @DataProvider(name = "provideUserConsentData")
+    public Object[][] provideUserConsentData() {
+        String authzCode = "67428657950009705658674645643";
+        String accessToken = "56789876734982650746509776325";
+        String idToken = "eyJzdWIiOiJQUklNQVJZXC9zdXJlc2hhdHQiLCJlbWFpbCI6InN1cmVzaGdlbXVudUBteW1haWwuY29tIiwibmFtZSI" +
+                "6IlN1cmVzaCBBdHRhbmF5YWtlIiwiZmFtaWx5X25hbWUiOiJBdHRhbmF5YWtlIiwicHJlZmVycmVkX3VzZXJuYW1lIjoic3VyZXN" +
+                "oZ2VtdW51IiwiZ2l2ZW5fbmFtZSI6IlN1cmVzaCJ9";
+
+        return new Object[][] {
```

Comment:
If possible add a comment saying what each combination corresponds to in the actual flow. Would be useful later if someone has to fix or add a new case

Reply:
These values are provided to cover all the branches in handleUserConsent() private method. Hard to bind the effect of the data on the main flow of  authorize() public method. In fact handleUserConsent() is method called from one branch of authorize() method to find the redirectUrl